### PR TITLE
Updated to arb-ts-v2

### PR DESCRIPTION
**Pull request by yahgwai**
* Updated to arb-ts-v2
* Improved schema validation errors with `better-ajv-errors`
* Now need to pass the l2 network id instead of the l1orl2

### CODE REVIEW

File: src/lib/instantiate_bridge.ts [hunk 1]
```diff
@@ -1,34 +1,41 @@
...
-    ? networks[network.partnerChainID]
-    : network;
-  const l2Network = networks[l1Network.partnerChainID];
-
-  const ethProvider = new providers.JsonRpcProvider(l1Network.rpcURL);
   const arbProvider = new providers.JsonRpcProvider(l2Network.rpcURL);
+  const ethProvider = new providers.JsonRpcProvider(l1Network.rpcURL)
 
   // random address for void signer:
   const l1Signer = new VoidSigner("0xAddA0B73Fe69a6E3e7c1072Bb9523105753e08f8", ethProvider)
```
Comment (DZGoldman) [inline src/lib/instantiate_bridge.ts:19]: Looks like we can do away with these signers altogether now

File: src/lib/token_list_gen.ts [hunk 2]
```diff
@@ -114,20 +122,20 @@ export const generateTokenList = async (
...
     } else {
-      console.log('no logo uri for ',token.l1TokenAddr, symbol);
+      console.log('no logo uri for ',token.token.l1TokenAddr, symbol);
       
     }
 
     return arbTokenInfo;
   }).filter((tokenInfo: ArbTokenInfo)=>{
-    return tokenInfo.extensions && tokenInfo.extensions.bridgeInfo[l1Network.chainID].originBridgeAddress !== "0x0000000000000000000000000000000000000001" 
+    return tokenInfo.extensions && tokenInfo.extensions.bridgeInfo[l1.network.chainID].originBridgeAddress !== "0x0000000000000000000000000000000000000001"
```
Comment (DZGoldman) [inline src/lib/token_list_gen.ts:131]: we can use the DISABLED_GATEWAY constant from arb-ts here

File: src/lib/token_list_gen.ts [hunk 3]
```diff
@@ -63,48 +63,56 @@ export const generateTokenList = async (
...
     logoUris.push(uri);
   }
 
-  let arbifiedTokenList:ArbTokenInfo[] = tokens.map((token, i: number) => {
-    const l2GatewayAddress = token.joinTableEntry[0].gateway.gatewayAddr;
-    const address = l2AddressesFromL1[i];
-    let { name:_name, decimals, symbol:_symbol } = tokenData[i];
+  let arbifiedTokenList:ArbTokenInfo[] = tokens
+      .map((t, i) => ({token: t, l2Address: l2AddressesFromL2[i], tokenDatum: tokenData[i]}))
+      .filter((t): t is typeof t & { l2Address: string } => t.l2Address != undefined && t.l2Address !== constants.AddressZero)
```
Comment (DZGoldman) [inline src/lib/token_list_gen.ts:96]: I might be misreading it, but isn't this filter redundant with the above

 ```tokens = tokens.filter((t, i) => l2AddressesFromL1[i] === l2AddressesFromL2[i])```
? 

Maybe add some comments here broadly explaining the map/filter/map levels? (Even if only to help me 😉 )

File: src/lib/utils.ts [hunk 4]
```diff
@@ -1,274 +1,222 @@
...
-    }
-    throw new Error('Data does not confirm to token list schema; not sure why');
+export const validateTokenListWithErrorThrowing = (
+  tokenList: ArbTokenList | TokenList,
+) => {
+  let valid = validateTokenList(tokenList)
+  if (valid) return true
+  else {
+    console.log('Invalid token list:')
+    throw new Error('Data does not confirm to token list schema; not sure why')
```
Comment (DZGoldman) [inline src/lib/utils.ts:129]: typo cofirm/conform (it was mine, but I'm just noticing here lol)

**Comment by DZGoldman** [comment]
Doesn't need to be in this PR, but I think we need to add a prettier config (so our linters are consistent)

**Comment by DZGoldman** [comment]
seeing compile time error: "Parameter 't' implicitly has an 'any' type."
I think there's some ts config / dependency issues to sort out

File: src/lib/instantiate_bridge.ts (hunk 1 above)
Comment (yahgwai) [reply src/lib/instantiate_bridge.ts:19]: :+1:

File: src/lib/token_list_gen.ts (hunk 2 above)
Comment (yahgwai) [reply src/lib/token_list_gen.ts:131]: updated

File: src/lib/token_list_gen.ts (hunk 3 above)
Comment (yahgwai) [reply src/lib/token_list_gen.ts:96]: I found that we got the zero address there sometimes, I guess it means that our graphql query may return some token addresses that havent been fully bridged. Maybe it's a syncing issue in the graph?

**Comment by yahgwai** [comment]
Yeah, I see from the diff I accidentally commited formatting changes. Woops! I'll add a prettier config soon.

File: src/lib/utils.ts (hunk 4 above)
Comment (yahgwai) [reply src/lib/utils.ts:129]: Fixed

**Comment by yahgwai** [comment]
> seeing compile time error: "Parameter 't' implicitly has an 'any' type."
> I think there's some ts config / dependency issues to sort out

I think you may need to checkout, build and link the latest from the arb-ts-next branch

**Review by DZGoldman — APPROVED** [review]
lgtm

**Pull request merged**